Polygon_mesh_processing/include/CGAL/Side_of_triangle_mesh.h

```diff
@@ -213,11 +232,19 @@ class Side_of_triangle_mesh
                                    faces(*tm_ptr).second,
                                    *tm_ptr, *opt_vpm);
           const_cast<AABB_tree_*>(tree_ptr)->build();
+#ifdef CGAL_HAS_THREADS
+        CGAL_SCOPED_LOCK(tree_mutex);
+        atomic_tree_ptr.store(tree_ptr, std::memory_order_release);
+#endif
         }
       }
-
+#ifdef CGAL_HAS_THREADS
+      return internal::Point_inside_vertical_ray_cast<GeomTraits, AABB_tree>()(
+            point, *atomic_tree_ptr.load(), ray_functor, vector_functor);
```

Comment:
This `load()` could be relaxed. It might be important for performances, because that `load()` is performed at each call to `operator()`.

Actually, even better for performance, I think you should use:
```suggestion
            point, *tree_ptr, ray_functor, vector_functor);
```